src/app/main/component/user/components/habit/add-new-habit/habit-invite-friends/habit-invite-friends-pop-up/habit-invite-friends-pop-up.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, OnInit } from '@angular/core';
+import { FriendArrayModel, FriendModel } from '@global-user/models/friend.model';
+import { UserFriendsService } from '@global-user/services/user-friends.service';
+
+@Component({
+  selector: 'app-habit-invite-friends-pop-up',
+  templateUrl: './habit-invite-friends-pop-up.component.html',
+  styleUrls: ['./habit-invite-friends-pop-up.component.scss']
+})
+export class HabitInviteFriendsPopUpComponent implements OnInit {
+  userId: number;
+  friends: FriendModel[];
+  allAdd = false;
+  addedFriends: FriendModel[] = [];
+
+  constructor(private userFriendsService: UserFriendsService) {}
+
+  ngOnInit() {
+    this.getUserId();
+    this.getFriends();
+  }
+
+  private getUserId() {
+    this.userId = +localStorage.getItem('userId');
+  }
+
+  getFriends() {
+    this.userFriendsService.getAllFriends(this.userId).subscribe((data: FriendArrayModel) => {
+      this.friends = data.page;
+    });
+  }
+
+  updateAllAdd() {
+    this.allAdd = this.friends !== null && this.friends.every((friend) => friend.added);
+  }
+
+  someAdd(): boolean {
+    if (this.friends == null) {
```

Comment:
what if this.friends will be undefined ?

Reply:
fixed